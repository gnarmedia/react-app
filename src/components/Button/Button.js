/**
 * @file Button component
 */

import React from 'react'
import PropTypes from 'prop-types'

/**
 * Renders Button component.
 *
 * @example
 * <Button>click me</Button>
 *
 * @param {*} children - Children.
 *
 * @returns {Function} Button component.
 */
const Button = ({ type, children }) => {
  const className = 'button is-block is-info is-large is-fullwidth'
  if (type === 'submit') {
    return (
      <button type="submit" className={className}>
        {children}
      </button>
    )
  }
  return (
    <button type="button" className={className}>
      {children}
    </button>
  )
}

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node.isRequired,
}

Button.defaultProps = {
  type: 'submit',
}

export default Button
