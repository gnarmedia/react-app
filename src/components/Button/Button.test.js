import React from 'react'
import { shallow } from 'enzyme'

import Button from './Button'

describe('Button', () => {
  it('should render the Button component as a submit', () => {
    const wrapper = shallow(<Button>click me</Button>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Button component as a button', () => {
    const wrapper = shallow(<Button type="button">click me</Button>)

    expect(wrapper).toMatchSnapshot()
  })
})
