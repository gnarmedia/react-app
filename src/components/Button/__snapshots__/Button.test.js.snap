// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button component as a button 1`] = `
<button
  className="button is-block is-info is-large is-fullwidth"
  type="button"
>
  click me
</button>
`;

exports[`Button should render the Button component as a submit 1`] = `
<button
  className="button is-block is-info is-large is-fullwidth"
  type="submit"
>
  click me
</button>
`;
